processing/src/main/java/io/druid/segment/IndexMerger.java

```diff
@@ -470,20 +480,71 @@ public String apply(@Nullable String input)
       }
     };
 
-    return makeIndexFiles(indexes, outDir, progress, mergedDimensions, mergedMetrics, null, rowMergerFn, indexSpec);
+    return makeIndexFiles(
+        indexes,
+        mergeAggregators(indexes),
+        outDir,
+        progress,
+        mergedDimensions,
+        mergedMetrics,
+        null,
+        rowMergerFn,
+        indexSpec
+    );
+  }
+
+  public static final AggregatorFactory[] mergeAggregators(List<IndexableAdapter> indexes)
+  {
+    Set<AggregatorFactory> mergedAggregators = Sets.newTreeSet(
+        Ordering.from(
+            new Comparator<AggregatorFactory>()
+            {
+              @Override
+              public int compare(AggregatorFactory a1, AggregatorFactory a2)
+              {
+                String name1 = a1.getName();
+                String name2 = a2.getName();
+                if (name1 == null) {
+                  return -1;
+                } else {
+                  return name1.compareTo(name2);
+                }
+              }
+            }
+        ).nullsFirst()
+    );
+    for (IndexableAdapter index : indexes) {
```

Comment:
also merge/reindex task have to fail in case auto merge of aggregators cant be done. For coordinator, we should sent an alert  in those situation and for person manually submitting the task it can just fail.
when reindexing is failing due to auto merge not possible, there will be a manual intervention and user can come and resolve the situation by explicitly doing a index task (to do reindexing) with explicit list of aggregators